src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java

```diff
@@ -1860,21 +1860,37 @@ else if ("test".equals(configName)) {
     }
 
     private static void validateChapterWiseTesting(Iterator<Node> itrChecks,
-          Iterator<Node> itrConfigs, Set<String> styleChecks, String styleName, String ruleName) {
+          Iterator<Node> itrSample, Set<String> styleChecks, String styleName, String ruleName) {
         boolean hasChecks = false;
+        final Set<String> checkLinks = new HashSet<>();
 
         while (itrChecks.hasNext()) {
             final Node module = itrChecks.next();
             final String moduleName = module.getTextContent().trim();
             final String href = module.getAttributes().getNamedItem("href").getTextContent();
             final boolean moduleIsCheck = href.startsWith("checks/");
 
+            final String partialConfigUrl = "https://github.com/search?q="
+                    + "path%3Asrc%2Fmain%2Fresources%20path%3A**%2F" + styleName;
+
             if (!moduleIsCheck) {
+                if (href.startsWith(partialConfigUrl)) {
+                    assertWithMessage("google_style.xml rule '" + ruleName + "' module '"
+                            + moduleName + "' has too many config links").fail();
+                }
                 continue;
             }
 
             hasChecks = true;
 
+            assertWithMessage("The module '" + moduleName + "' in the rule '" + ruleName
+                    + "' of the style guide '" + styleName
+                    + "_style.xml' should not appear more than once in the section.")
+                    .that(checkLinks)
+                    .doesNotContain(href);
+
+            checkLinks.add(href);
```

Comment:
This is to ensure that a module does not appear twice in the section, I'm storing module links in the Set instead of Module name from:

https://github.com/checkstyle/checkstyle/blob/bf1f739e7926b857b12a4df78825bd667fe59fac/src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java#L1796

storing check link will ensure that even if a contributor mis-types the Check name, we're able to detect duplicate module usage through its link.

Reply:
We were checking duplicate module appearance before. I thought it would be great to cover this edge case as well